hotness/buildsys.py

```diff
@@ -112,33 +112,27 @@ def scratch_build(self, session, name, source):
             name=name, url=self.url_for(task_id)))
         return task_id
 
-    def run(self, cmd, cwd=None):
-        _log.info("Running %r in %r", cmd, cwd)
-        p = sp.Popen(cmd, cwd=cwd, stdout=sp.PIPE, stderr=sp.PIPE)
-        out, err = p.communicate()
-        if out:
-            _log.debug(out)
-        if err:
-            _log.warning(err)
-        if p.returncode != 0:
-            message = 'cmd:  %s\nreturn code:  %r\nstdout:\n%s\nstderr:\n%s'
-            raise Exception(message % (' '.join(cmd), p.returncode, out, err))
-        return out
-
     def handle(self, package, upstream, version, rhbz):
         """ Main API entry point.
 
         Bumps the version on a package and requests a scratch build.
 
-        Returns the task_id of the scratch build.
+        Args:
+            package (str): The package name.
+            upstream (str): The new upstream version.
+            version (str): Unused, exists for API compatibility at the moment.
+            rhbz (?): The bugzilla object with a ``bug_id`` attribute.
+
+        Returns:
+            tuple: A tuple of (koji task ID, patch path, BZ attachment name).
         """
```

Comment:
Yeah, although the API needs to change anyway. Some of the arguments aren't even used :(